internal/cmd/commands/authenticate/authenticate.go

```diff
@@ -35,5 +48,42 @@ func (c *Command) Help() string {
 }
 
 func (c *Command) Run(args []string) int {
-	return cli.RunResultHelp
+	client, err := c.Client(base.WithNoTokenScope(), base.WithNoTokenValue())
+	if c.WrapperCleanupFunc != nil {
+		defer func() {
+			if err := c.WrapperCleanupFunc(); err != nil {
+				c.PrintCliError(fmt.Errorf("Error cleaning kms wrapper: %w", err))
+			}
+		}()
+	}
+	if err != nil {
+		c.PrintCliError(fmt.Errorf("Error creating API client: %w", err))
+		return base.CommandCliError
+	}
+
+	// Lookup the primary auth method ID in the global scope
+	aClient := authmethods.NewClient(client)
+	pri, err := getPrimaryAuthMethodId(c.Context, aClient, "global", "")
+	if err != nil {
+		c.PrintCliError(errors.New("Error looking up primary auth method ID for global scope. Try setting primary auth method for global scope, or use a auth method subcommand (see 'boundary authenticate -h' for available sub command usage)."))
+		return base.CommandUserError
+	}
+
+	c.FlagAuthMethodId = pri
+
+	switch {
+	case strings.HasPrefix(c.FlagAuthMethodId, "ampw"):
+		cmd := PasswordCommand{Command: c.Command}
+		cmd.Run([]string{})
+
+	case strings.HasPrefix(c.FlagAuthMethodId, "amoidc"):
```

Comment:
I was under the impression that we wanted to remove these prefixes from IDs at some point, would it not be better to check the type of the primary auth method directly?

Reply:
Ah, good catch, we can definitely change to verifying by type. I'll get that taken care of.

Reply:
That said it'd be better to use the various consts -- and in some of those cases perhaps moving those consts to the globals package.

Reply:
👍 Fixed